### PR TITLE
Document that a different plugin checks references

### PR DESCRIPTION
**Pull request by dlipovetsky**
Update the documentation, so that it's clear that a different plugin checks references. When I read the doc myself, that wasn't clear, and I mistakenly filed #52 against this plugin. 

/cc @ChristianMurphy

### CODE REVIEW

**Comment by dlipovetsky** [comment]
The build failed because of lint issues :joy:. I'll fix.

File: readme.md [hunk 1]
```diff
@@ -96,11 +98,11 @@ Say we have the following file, `example.md`:
 
 This [exists](#alpha).
 This [one does not](#does-not).
-References and definitions are [checked][alpha] [too][charlie].
```
[the remark this answers is not in the document]
Comment (dlipovetsky) [reply readme.md:99]: Could you please share an example that the linter will reject?

[the remark this answers is not in the document]
Comment (ChristianMurphy) [reply readme.md:99]:
```markdown
[defined-but-dne][]

[defined-but-dne]: #not-a-real-header

[the remark this answers is not in the document]
Comment (dlipovetsky) [reply readme.md:99]: Excellent. Thank you. I'll add that.

[the remark this answers is not in the document]
Comment (dlipovetsky) [reply readme.md:99]: Added

**Comment by dlipovetsky** [comment]
The example, for some reason, said that headers in other files are not checked. But they are! I fixed that. LMK if I misunderstood something yet again :grimacing: .

**Comment by dlipovetsky** [comment]
Thanks for the help @ChristianMurphy

File: readme.md [hunk 2]
```diff
@@ -10,8 +10,10 @@
 
 [**remark**][remark] plugin to validate that Markdown links and images reference
 existing local files and headings.
-It does not check external URLs (see [`remark-lint-no-dead-urls`][no-dead-urls]
-for that).
+It does not check:
+
+*   External URLs (see [`remark-lint-no-dead-urls`][no-dead-urls])
+*   References (see [`remark-lint-no-undefined-references`][no-undef-refs])
```
Comment (wooorm) [inline readme.md:16]:
```suggestion
*   References without definitions (see [`remark-lint-no-undefined-references`][no-undef-refs])
```

File: readme.md [hunk 3]
```diff
@@ -94,14 +96,17 @@ Say we have the following file, `example.md`:
...
-References and definitions are [checked][alpha] [too][charlie].
 
 # Bravo
 
-Headings in `readme.md` are [not checked](readme.md#bravo).
-But [missing files are reported](missing-example.js).
+Headings in `readme.md` are [checked](readme.md#nosuchheading).
+And [missing files are reported](missing-example.js).
+
+References are also checked:
```
Comment (wooorm) [inline readme.md:109]:
```suggestion
Definitions are also checked:
```

File: readme.md [hunk 4]
```diff
@@ -94,14 +96,17 @@ Say we have the following file, `example.md`:
...
 
 [alpha]: #alpha
 [charlie]: #charlie
```
Comment (wooorm) [inline readme.md:112]:
```suggestion
[charlie]: #charlie

References w/o definitions are not checked: [delta]
```

**Comment by wooorm** [comment]
@dlipovetsky Friendly ping!

**Comment by wooorm** [comment]
@dlipovetsky Ping! 🛎😊

**Comment by wooorm** [comment]
Landed the changes myself, thanks for your initial work @dlipovetsky!

**Pull request closed without merging**